### DISCUSSION TITLE
Running vscode in WSL-2 podman machine or podman container

### DISCUSSION THREAD

**Discussion by mwoodpatrick** [containers/podman#14571] [Q&A]
I would appreciate any pointers on running vscode in WSL-2 podman machine or podman container

I can get vscode to conect to the wsl-2 podman machine but it has the user set to root even though I set the default user to 'user' in /etc/wsl.conf in the distro (using vi)

Is there a way to set and query config values within the distro equivalent of the ubuntu function:

      ubuntu config --default-user <user>

https://www.tenforums.com/tutorials/128152-set-default-user-windows-subsystem-linux-distro-windows-10-a.html#:~:text=Set%20Default%20User%20to%20Specified%20User%20for%20a,user%20for%2C%20and%20press%20Enter.%20%28see%20screenshot%20below%29

Is there a way to specify the username that's used in the podman machine to something other than 'user'?

Any pointers on using vscode to access a container created by podman would be appreciated. I do see there is discussion of this in 

     https://github.com/microsoft/vscode-docker/issues/3263

Is there a way to get it to work before this issue is resolved? 

Any pointers would be much appreciated.

**Comment by rhatdan** [comment]
@n1hility PTAL

**Comment by n1hility** [comment]
@mwoodpatrick which mode of remote vscode are you trying? This one: https://code.visualstudio.com/docs/remote/wsl ?

Due to https://github.com/microsoft/vscode-remote-release/issues/286 this won't work yet. I will need to investigate whats possible since systemd is used in a nested namespace, which may cause other issues with this extension. 

Note: that the wsl wrapper will currently break if the wsl init config uses a non-root user (root is needed to setup systemd).  That I will fix.

The container or ssh approach should function. In the container approach it will use 'user' if rootless is selected (the default)

**Reply by mwoodpatrick** [reply]
Any updates on this? I would appreciate any pointers on using current best practices on using podman with MS code-server and vscide in general
